src/averageAges.js

```diff
@@ -37,7 +39,12 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f' && (!withChildren || people.some(child => child.mother === person.name)));
+
+  const ages = women.map(woman => woman.died - woman.born);
+  const totalAge = ages.reduce((sum, age) => sum + age, 0);
```

Comment:
you can apply method chaining 
```suggestion
  const totalAge = women.map(woman => woman.died - woman.born).reduce((sum, age) => sum + age, 0);
```